js/src/forum/extenders/extendDiscussionListState.tsx

```diff
@@ -8,7 +8,11 @@ export default function extendDiscussionListState() {
       const negate = app.discussions.bestAnswer === '2';
       const prepend = negate ? '-' : '';
 
-      params.filter[`${prepend}solved-discussions`] = true;
+      if (!params.filter) {
+        params.filter = {};
+      }
+
+      (params.filter as Record<string, string>)[`${prepend}solved-discussions`] = 'true';
```

Comment:
If we import the `PaginatedListParams` type and use that for the `params` parameter, we can save ourselves this gymnastic and leave it like it was before

```tsx
...
import type { PaginatedListParams } from 'flarum/common/states/PaginatedListState';

export default function extendDiscussionListState() {
  extend(DiscussionListState.prototype, 'requestParams', function (params: PaginatedListParams) {
```